Source/Classes/PINRemoteImageManager.h

```diff
@@ -542,12 +542,26 @@ typedef void(^PINRemoteImageManagerProgressDownload)(int64_t completedBytes, int
  Cancel a download. Canceling will only cancel the download if all other downloads are also canceled with their associated UUIDs. 
  Canceling *does not* guarantee that your completion will not be called. You can use the UUID provided on the result object to verify
  the completion you want called is being called.
+ @param UUID NSUUID of the task to cancel.
  @param storeResumeData if YES and the server indicates it supports resuming downloads, downloaded data will be stored in the memory
  cache and used to resume the download if the same URL is attempted to be downloaded in the future.
  */
 - (void)cancelTaskWithUUID:(nonnull NSUUID *)UUID storeResumeData:(BOOL)storeResumeData;
 
 /**
+ Cancel a download if the estimated time to complete the download exceeds the passed in value.
+ Canceling will only cancel the download if all other downloads are also canceled with their associated UUIDs.
+ Canceling *does not* guarantee that your completion will not be called. You can use the UUID provided on the result object to verify
+ the completion you want called is being called.
+ @param UUID NSUUID of the task to cancel.
+ @param timeToExceed only cancel the download if the expected time to complete it is greater than timeToExceed
+ @param storeResumeData if YES and the server indicates it supports resuming downloads, downloaded data will be stored in the memory
+ cache and used to resume the download if the same URL is attempted to be downloaded in the future.
+ */
+
+- (void)cancelTaskWithUUID:(nonnull NSUUID *)UUID ifExpectedToExceedTimeInterval:(NSTimeInterval)timeToExceed storeResumeData:(BOOL)storeResumeData;
```

Comment:
This time interval is assumed to be relative to "now" when it's provided but won't be evaluated until some (small) time in the future.

Might be better (clearer and more predictable) for this to take an absolute "deadline" time beyond which any longer-running tasks are cancelled.

I know precision here is sloppy because it's all based on estimates, but it still might be a better API overall.